### PR TITLE
Add missing plugin hooks to docs

### PR DESCRIPTION
**Pull request by altendky**
pytest-dev/pytest#4896

- [ ] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.

### CODE REVIEW

**Comment by altendky** [comment]
As mentioned in #4896, I'm really not confident where these additions should go.  So, tear it apart...

File: doc/en/reference.rst [hunk 1]
```diff
@@ -629,12 +634,15 @@ items, delete or otherwise amend the test items:
 
 .. autofunction:: pytest_collection_modifyitems
 
+.. autofunction:: pytest_collection_finish
```
Comment (nicoddemus) [inline doc/en/reference.rst]: They all seem properly placed to me, good work!

**Comment by nicoddemus** [comment]
Anything else you feel needs adding to this @altendky? Otherwise I think we should change from "WIP" status and merge this. 👍

**Comment by altendky** [comment]
@nicoddemus, ah well, I totally missed your comment.  Sorry.

**Comment by altendky** [comment]
@nicoddemus, do you want a changelog entry?

**Comment by nicoddemus** [comment]
> @nicoddemus, do you want a changelog entry?

Don't think we need for this case. Thanks! Merging.

**Pull request merged**